src/test/java/org/folio/test/support/matcher/MappableMatcher.java

```diff
@@ -0,0 +1,38 @@
+package org.folio.test.support.matcher;
+
+import java.util.function.Function;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+public final class MappableMatcher<T, V> extends TypeSafeDiagnosingMatcher<T> {
```

Comment:
What is a mappable matcher? What does mappable mean in this context?

Reply:
This matcher is used to map something to a value and then assert this mapped value. 

For example:

Jayway `hasJsonPath` matchers can assert json as string only, however we usually use `JsonObject`. We can now use following logic in order to assert an JsonObject with string `hasJsonPath` matcher:

```
assertThat(jsonObject, new MappableMatcher<JsonObject, String>(JsonObject::toString,
allOf(
   hasJsonPath(...),
   hasJsonPath(...)
)));
```

Reply:
This converts the `JsonObject` to a `String` and then delegates the rest of the matching to other matchers?

Ok, maybe this could be renamed `TypeMappingMatcher` or it could be made to be more specific to the JSON Path use case and renamed JSONPathMatcher?

Reply:
Renamed to TypeMappingMatcher